skxray/roi.py

```diff
@@ -229,3 +231,87 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
     edges = np.cumsum(steps).reshape(-1, 2)
 
     return edges
+
+
+def divide_pies(image_shape, radius, calibrated_center,
+                      num_angles, rotate='N',):
+    """
+    This function will provide the indices when a circular roi
+    is divided into pies.
+
+    Parameters
+    ----------
+    image_shape : tuple
+        shape of the image (detector X and Y direction)
+        Order is (num_rows, num_columns)
+
+    radius : float
+        radius of the circle
+
+    calibrated_center : tuple
+        defining the center of the image
+        (column value, row value) (mm)
+
+    num_angles : int
+        number of angles ring divide into
+        angles are measured from horizontal-anti clock wise
+
+    rotate : {'Y', 'N'}, optional
+        to make angles measured from vertical-anti clock wise
+
+    Returns
+    -------
+    labels_grid : array
+        indices of the required roi's
+        shape is ([image_shape[0], image_shape[1]])
+    """
+    angle_grid  = get_angle_grid(image_shape, calibrated_center)
+    # required angles
+    angles = np.linspace(0, 360, num_angles)
+    # the indices of the bins(angles) to which each value in input
+    #  array(angle_grid) belongs.
+    ind_grid = (np.digitize(np.ravel(angle_grid), angles,
+                            right=False)).reshape(image_shape)
+
+    # radius grid for the image_shape
+    grid_values = core.pixel_to_radius(image_shape, calibrated_center)
+
+    if rotate == 'Y':
+        ind_grid = np.rot90(ind_grid)
+
+    ind_grid[grid_values > radius] = 0
+
+    labels_grid = ind_grid.reshape(image_shape)
+
+    return labels_grid
+
+
+def get_angle_grid(image_shape, calibrated_center):
+    """
+    This function will provide angle values for the whole grid
+    from the calibrated center
+
+    Parameters
+    ----------
+    image_shape : tuple
+        shape of the image (detector X and Y direction)
+        Order is (num_rows, num_columns)
+
+    calibrated_center : tuple
+        defining the center of the image
+        (column value, row value) (mm)
+
+    Returns
+    -------
+    angle_grid : array
+        angle values from the calibrated center
+        shape image_shape
+    """
```

Comment:
@sameera2004 and I benchmarked this against trackpy's theta_mask (https://github.com/soft-matter/trackpy/blob/master/trackpy/masks.py#L40) and found theta_mask to be 30X times faster. Note that theta_mask assumes r=0 is in the center of the image, which may be the source of perf gains.